pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -394,10 +419,17 @@ func calculateStatus(config *appsapi.DeploymentConfig, rcs []*v1.ReplicationCont
 		unavailableReplicas = 0
 	}
 
+	var generation int64
```

Comment:
get rid of "else" with:

```go
generation := config.Status.ObservedGeneration
if updateObservedGeneration {
  generation = config.Generation
}
```

Reply:
getting rid of else would mean two writes instead of one, but I can change it if you want

Reply:
write is fine, else is harder to brain scan :)

Reply:
done